sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeProjection.scala

```diff
@@ -192,26 +192,30 @@ object GenerateUnsafeProjection extends CodeGenerator[Seq[Expression], UnsafePro
     val fixedElementSize = et match {
       case t: DecimalType if t.precision <= Decimal.MAX_LONG_DIGITS => 8
       case _ if ctx.isPrimitiveType(jt) => et.defaultSize
-      case _ => 0
+      case _ => 8
```

Comment:
please comment about why it's 8, e.g. we store offset and length for the element blabla